compiler/plugins/target/AMD-AIE/iree-amd-aie/IR/AMDAIEOps.td

```diff
@@ -1212,6 +1212,11 @@ def AMDAIE_LogicalObjectFifoFromBuffersOp
 
     // Return the locks on the producer side (write) on the requested tile.
     llvm::SmallVector<LockOp> getProducerLocksOnTile(TileOp tileOp);
+
+    // A utility to replace this logical objectFifo operation with a new one 
+    // with new tiles.
+    FailureOr<LogicalObjFifoOpInterface> replaceWithNewTiles(
+      ::mlir::RewriterBase &rewriter, ::mlir::ValueRange tiles);
```

Comment:
If it's already declared in the Interface `LogicalObjFifoOpInterface`, wouldn't these already be part of the ops inheriting them?

What I mean to ask is if the change in `AMDAIEOps.td` is necessary.

Reply:
The methods need to be declared in the ops that implement the interface, but I was able to simplify it with: `DeclareOpInterfaceMethods<LogicalObjFifoOpInterface, ["replaceWithNewTiles"]>`